httpcore/_async/connection.py

```diff
@@ -1,54 +1,51 @@
 from ssl import SSLContext
-from typing import Optional, Tuple, cast
+from typing import Optional, Tuple
 
-from .._backends.auto import AsyncBackend, AsyncLock, AsyncSocketStream, AutoBackend
-from .._exceptions import ConnectError, ConnectTimeout
+from .._backends.auto import AsyncBackend, AsyncSocketStream, AutoBackend
 from .._types import URL, Headers, Origin, TimeoutDict
-from .._utils import exponential_backoff, get_logger, url_to_origin
-from .base import (
-    AsyncByteStream,
-    AsyncHTTPTransport,
-    ConnectionState,
-    NewConnectionRequired,
-)
+from .._utils import get_logger, url_to_origin
+from .base import AsyncByteStream, AsyncHTTPTransport, ConnectionState
 from .http import AsyncBaseHTTPConnection
 from .http11 import AsyncHTTP11Connection
 
 logger = get_logger(__name__)
 
-RETRIES_BACKOFF_FACTOR = 0.5  # 0s, 0.5s, 1s, 2s, 4s, etc.
-
 
 class AsyncHTTPConnection(AsyncHTTPTransport):
     def __init__(
         self,
         origin: Origin,
-        http2: bool = False,
-        uds: str = None,
+        socket: AsyncSocketStream,
```

Comment:
This is the change from which all others derive: `socket` becomes required, and will be passed by whoever wants to interact with an HTTP connection (in our case, the connection pool).